### PR TITLE
Add checkbox to stop player after queue is empty

### LINKED ISSUE #43: rfe: stop playing once queue runs out
Original [issue 43](https://code.google.com/p/quodlibet/issues/detail?id=43) created by hhaamu on 2008-11-03T18:33:54.000Z:

QL currently has a &quot;Stop after this song&quot; feature hidden beneath the play
button. I'd like it to be duplicated for the queue feature. I often queue
up stuff and wish QL to stop playing after they are played.

Currently, to accomplish this I must either a) clear the playlist or b)
wait for the last song in the queue and use the aforementioned feature.

### PR DESCRIPTION
**Pull request by frestr**
Fixes #43

Although the "Queue Only" plugin has this as a side effect, the effect of always adding songs to the queue may not be desired. This addition instead adds a checkbox - "Stop After Empty" - next to the "Random" checkbox that has no other side effects than stopping the player after the queue is empty, instead of continuing to play a song from the song pane.

### CODE REVIEW

File: quodlibet/quodlibet/qltk/queue.py [hunk 1]
```diff
@@ -121,16 +123,27 @@ def __init__(self, library, player):
...
+                _("_Random"), "memory", "shufflequeue")
+        rand_checkbox.connect('toggled',
+                              self.__queue_shuffle,
+                              self.queue.model)
+        rand_checkbox.set_active(config.getboolean("memory", "shufflequeue"))
+        rand_checkbox.set_no_show_all(True)
+        left.pack_start(rand_checkbox, False, True, 0)
+
+        stop_checkbox = ConfigCheckButton(
+            _("Stop After Empty"), "memory", "queue_stop_after_empty")
```
Comment (declension) [inline quodlibet/quodlibet/qltk/queue.py]: I think _Stop Once Empty_ reads a bit better than _Stop After Empty_

**Comment by declension** [comment]
Looks reasonable in general, and works for me - thanks.

Part of me still thinks this can / should be solved as a play order plugin / configuration for the queue itself, but that's a much bigger piece so maybe best to get this in for now.

File: quodlibet/quodlibet/qltk/queue.py (hunk 1 above)
Comment (frestr) [inline quodlibet/quodlibet/qltk/queue.py]: I agree. I based it on "Stop After This Song" under Controls, but the wording makes more sense in that case.

**Comment by frestr** [comment]
I thought about making it a play order plugin, but that doesn't make much sense the way I see it, as it doesn't really have anything to do with the play order.

I think having a separate configuration window for the queue would be nice, especially if we later want to add even more queue options not relating to the play order. Perhaps we could add a new "Queue" tab in the File->Preferences window?

**Comment by declension** [comment]
Yep I think all it's a bit complicated conceptually and UI-wise both now and maybe forever. Essentially I wanted to provide a way to abstract the `PlaylistMux` better, and also unify the "Random" with Shuffle plugins (why can't the queue be shuffled in a weighted way, or whatever).

That way, this and the Queue only plugin could live somewhere a bit more meaningful. Though of course that then makes it _more_ complicated, if slightly neater / more powerful.

...but again this is a whole separate piece really.

@frestr if you can make that text change, then assuming @lazka doesn't have anything I think it's good to go.

**Comment by declension** [comment]
Thanks

**Comment by lazka** [comment]
Thanks

**Pull request merged**